src/qvs.cpp

```diff
@@ -0,0 +1,316 @@
+      auto& c = csr[i];
+      // If there was no fixed file specified in input
+      if (in.getFixed().empty() && i % 3 == 0){
+      // Initialize the csr objects with theta-dtheta, theta, theta+dtheta in order 
+      // to compute adrFixed if no file was specified in input.
+        c.init();
+        switch (counter % 3) { 
+        case 0:
+            cout << "theta - dtheta auxiliary density response"
+            << " fixed component stored in file" << endl;
+            break;
+        case 1:
+            cout << "theta fixed auxiliary density response" 
+            << " component stored in file"<< endl;
+            break;
+        case 2:
+            cout << "theta + dtheta auxiliary density response" 
+            << " fixed component stored in file" << endl;
+            break;
+    }
+        counter++;
+        // If file was specified read corresponding adrFixed from file 
+      } else { 
+          const double theta = in.getDegeneracy();
+          // Create a stringstream to help with formatting
+          std::stringstream ss; 
+          // Set precision to 3rd decimal
+          ss << std::fixed << std::setprecision(3); 
+          ss << "adr_fixed_theta" << theta << "_matsubara" << in.getNMatsubara() << ".bin";
+          std::string filename = ss.str();
+          // Compute ssfHF 
+          c.Rpa::computeChemicalPotential();
+          c.Rpa::computeIdr();
+          c.Rpa::computeSsfHF();
+          // Read adrFixed from file
+          c.readAdrFixedFile(c.adrFixed, filename, false);
+      }
+    }
+	  csrIsInitialized = true;
+    }
+    doIterations();
+    computed = true;
+    return 0;
+  }
+  catch (const std::runtime_error& err) {
+    std::cerr << err.what() << std::endl;
+    return 1;
+  }
+}
+  
+void QStructProp::doIterations() {
+  const auto& in = csr[0].in;
+  const int maxIter = in.getNIter();
+  const int ompThreads = in.getNThreads();
+  const double minErr = in.getErrMin();
+  double err = 1.0;
+  int counter = 0;
+  // Define initial guess
+  for (auto& c : csr) { c.initialGuess(); }
+  // Iteration to solve for the structural properties
+  const bool useOMP = ompThreads > 1;
+  while (counter < maxIter+1 && err > minErr ) {
+    #pragma omp parallel num_threads(ompThreads) if (useOMP)
+    {
+      #pragma omp for
+      for (auto& c : csr) {
+        c.computeAdrStls();
+      }
+      #pragma omp for
+      for (size_t i = 0; i < csr.size(); ++i) {
+        auto& c = csr[i];
+        c.computeAdr();
+        c.computeSsf();
+        if (i == RS_THETA) {err = c.computeError(); }
+        c.updateSolution(); 
+      }
+    counter++;
+    }
+  }
+  printf("Alpha = %.5e, Residual error "
+	 "(structural properties) = %.5e\n", csr[RS_THETA].alpha, err);
+  // Set static structure factor for output
+  for (auto& c : csr) { c.ssf = c.ssfOld; }
+}
+
+vector<double> QStructProp::getQ() const  {
+  return getBase([&](const QStlsCSR& q) -> double {
+    return q.getQAdder();
+  });
```

Comment:
Not sure that this parenthesis belongs here

Reply:
If I remove it the code doesn't compile. I think it's needed because of the double return statement (not exactly sure).

Reply:
True, my bad!